### ISSUE TITLE
Repo.Smart#exists()

### ISSUE THREAD

**Issue by yegor256** [jcabi/jcabi-github#1379]
Let's add a method `exists()` to `Repo.Smart` to check whether the repository exists or not.

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) please, pay attention to this issue

**Comment by yegor256** [comment]
@amihaiemil I would do it similar to `RtIssue.exists()`. Also, I would move `RtIssue.exists()` to `Issue.Smart`. Can't understand why it's in the `Issue` interface :(

**Comment by yegor256** [comment]
@amihaiemil also, `User.Smart#exists()` is very similar. I don't understand why those methods are different in implementation. Maybe we should introduce something like `Existence` and use it everywhere?

```java
interface User {
  class Smart {
    public boolean exists() throws IOException {
      return new Existence(this.jsonReadable).check();
    }
  }
}
```

**Comment by yegor256** [comment]
@amihaiemil I guess, we need those `exists()` methods here:

  * `Blob.Smart`
  * `Branch.Smart`
  * `Comment.Smart`
  * `Commit.Smart`
  * `Event.Smart`
  * `Fork.Smart`
  * `Gist.Smart`
  * `GistComment.Smart`
  * `Hook.Smart`
  * `Issue.Smart`
  * `Label.Smart`
  * `Milestone.Smart`
  * `Organization.Smart`
  * `PublicKey.Smart`
  * `Pull.Smart`
  * `Release.Smart`
  * `Repo.Smart`
  * `Tag.Smart`
  * `User.Smart`

**Comment by amihaiemil** [comment]
@yegor256 yes, sounds like a good idea; do you want to do it?

**Comment by 0crat** [comment]
Job [#1379](https://github.com/jcabi/jcabi-github/issues/1379) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256)

**Comment by amihaiemil** [comment]
@llorllale want to do this one?

**Comment by 0crat** [comment]
The job [#1379](https://github.com/jcabi/jcabi-github/issues/1379) assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/C9R53K5JA/971345746); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html)

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by 0pdd** [comment]
@yegor256 the puzzle [#1382](https://github.com/jcabi/jcabi-github/issues/1382) is still not solved.

**Comment by amihaiemil** [comment]
@yegor256 I think this can be closed

**Comment by yegor256** [comment]
@amihaiemil can you please release it?

**Comment by amihaiemil** [comment]
@yegor256 it is released since yesterday, version 0.38 :)

**Comment by 0crat** [comment]
Order was finished: +30 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
The job [#1379](https://github.com/jcabi/jcabi-github/issues/1379) is now out of scope

**Comment by 0pdd** [comment]
@yegor256 all 2 puzzles are solved here: [#1382](https://github.com/jcabi/jcabi-github/issues/1382), [#1384](https://github.com/jcabi/jcabi-github/issues/1384).
